lib/app_web/controllers/page_controller.ex

```diff
@@ -1,7 +1,58 @@
 defmodule AppWeb.PageController do
   use AppWeb, :controller
+  alias App.Ctx.Person
+  alias App.Ctx
+
+  # auth_controller assign current_person on each request
+  # when the person is loggedin, otherwise the value doesn't exists or is nil
+  def index(%{assigns: %{current_person: person}} = conn, _params) when not is_nil(person) do
+    redirect(conn, to: Routes.person_path(conn, :info))
+  end
 
   def index(conn, _params) do
-    render(conn, "index.html")
+    url_oauth_google = ElixirAuthGoogle.generate_oauth_url(conn)
+    changeset = Person.changeset_registration(%Person{}, %{})
+    render(conn, "index.html",
+      [url_oauth_google: url_oauth_google, changeset: changeset])
+  end
+
+
+  def register(conn, %{"person" => person_params}) do
+    person = Ctx.get_person_by_email(person_params["email"])
+
+    if is_nil(person) do #create new person
+      case Ctx.register_person(person_params) do
+        {:ok, person} ->
+          create_basic_session(conn, person)
+
+        {:error, %Ecto.Changeset{} = changeset} ->
+          url_oauth_google = ElixirAuthGoogle.generate_oauth_url(conn)
```

Comment:
this looks very similar to the `index` function above. could we just invoke that?

Reply:
If we redirect to the index page we will loose the errors information contained in the changeset and the error won't be displayed to the user